cutils.h

```diff
@@ -356,6 +357,81 @@ static inline void inplace_bswap32(uint8_t *tab) {
     put_u32(tab, bswap32(get_u32(tab)));
 }
 
+static inline double fromfp16(uint16_t v) {
+    double d, s;
+    int e;
+    if ((v & 0x7C00) == 0x7C00) {
```

Comment:
Personal preference probably, but I'd define this like FP6_INF or something, for readability...

Reply:
I thought about that but... for doubles there's `INFINITY` from math.h and you can write `-INFINITY` to get a negative infinity. For FP16_INFINITY that doesn't work because -0x7C00 is 0x8400 and that's a regular number. I don't like that lack of parity.